gen/field.go

```diff
@@ -100,12 +103,36 @@ func (f fieldGroupGenerator) DefineStruct(g Generator) error {
 			// optional collections because omitempty doesn't differentiate
 			// between nil and empty collections.
 
+			jsonTags := ""
+			jsonFieldName := f.Name
+
 			if (isStructType(f.Type) || isPrimitiveType(f.Type)) && !f.Required {
-				return fmt.Sprintf("`json:\"%s,omitempty\"`", f.Name)
+				jsonTags += ",omitempty"
+			}
+
```

Comment:
This function is complicated enough at this point to pull it into a separate method.

    TemplateFunc("tag", f.generateTag)